### DISCUSSION TITLE
When is a release version expected?

### DISCUSSION THREAD

**Discussion by davidop** [microsoft/PowerPlatform-DataverseServiceClient#72] [Q&A] [answered]
We are currently migrating some Azure Functions from net462 to netcoreapp3.1 and we are very interested in this nuget package.

**Comment by MattB-msft** [comment, marked as answer]
We are still a ways away from a release version of this. 
Major updates pending: 
   We will soon be pushing updated to move to MSAL from AAD,
   Rename / renamespacing to new names for CDS
   Migration of more messages to underlying webAPI transport.

**Comment by davidop** [comment]
A shame.  Any suggestions for working with Azure Functions and CRM?

**Comment by jordimontana82** [comment]
Hi @MattB-msft , I understand there's lots of underlying changes and it's not a trivial task at all to estimate when a release version will be available, but do you have a general very high level estimate of when that could happen like (2021 Q1, 2022, ... ?) 
: )

Thanks!

**Reply by MattB-msft** [reply]
We are hopeful that the last major exterior change will occur with the next rename.  Aside from that we expect that the public interface will switch over to semantic versioning post that. 
However there will still be significant work inside the the client as we move over to the webAPI.
We are hoping to be able to finalize this work before the end of 2021,  but right now I cannot be more definitive then that.   

We will remove the preview tag from it before then, but still refer to it as a 0.X.X release. 

MattB

**Reply by jordimontana82** [reply]
Ok, thanks for the update Matt, much appreciated.

**Reply by luisgmgouveia** [reply]
Hi @MattB-msft , I understand you're still working on it, but is there any update on this since the 15/dec/2020? When are you planning to remove the preview tag? Thank you!

**Reply by luisgmgouveia** [reply]
Hi @MattB-msft , can you please answer my previous question (21 days ago)? Many thanks!

**Reply by davidop** [reply]
Hi Luis! I am using the package https://github.com/microsoft/PowerPlatform-DataverseServiceClient/releases/tag/0.4.12 for my azure functions in netcoreapp3.1

**Reply by MattB-msft** [reply]
The preview tag was removed from the Nuget packages already.
